pkg/agent/agent.go

```diff
@@ -72,14 +72,16 @@ func init() {
 
 type AgentOptions struct {
 	RegistrationAgent *spoke.SpokeAgentOptions
-	KubeConfig        *rest.Config
-	eventRecorder     events.Recorder
 
 	StatusSyncInterval                     time.Duration
 	AppliedManifestWorkEvictionGracePeriod time.Duration
 
 	Burst int
 	QPS   float32
+
+	KubeConfig string
+
+	eventRecorder events.Recorder
```

Comment:
can we capitalize `eventRecorder` so that it can be used outside?

Reply:
I don't think we should use this outside, this recorder has component name, it should only use for the this agent